[H]154.cpp

```diff
@@ -0,0 +1,34 @@
+// Problem: 154. Find Minimum in Rotated Sorted Array II - Hard
+// Link: https://leetcode.com/problems/find-minimum-in-rotated-sorted-array-ii/
+
+/* Nhận xét:
+- Xử lý như bình thường:
+   + Nếu mid > r => (mid, r] là đoạn bất thường, điểm nhỏ nhất nằm ở đây. Xét tiếp đoạn [l = mid + 1, r]
+   + Nếu ngược lại, => [l, mid] là đoạn bất thường. Xét tiếp đoạn [l, r = mid]. Chú ý ở đây phải là r = mid, vì mid có thể = r => mid cũng có thể là điểm nhỏ nhất
+   => Gán r = mid để không bỏ sót điểm mid.
+   + Ta đặt điều kiện (l - r) > 0 => khi kết thúc thì l = r => return l hay r đều ok.
+- Trường hợp đặc biệt: l = mid = r => không thể biết bỏ đoạn nào => bắt buộc phải xét tiếp cả 2, ta chỉnh lại 2 biên l++ và r--.
+*/
+
+class Solution {
+public:
+    int findMin(vector<int>& nums) {
+        int l = 0, r = nums.size() - 1;
```

Comment:
ok. Ổn cả về coding style lẫn algo. Bài này chú có tham khảo sol ko thế? :))

Reply:
Thi thố mới cheat chứ học hành thì phải đàng hoàng chứ :))
Nói chung anh đều giải trước rồi mới xem discussion. Trường hợp mà ko giải nổi phải xem discussion thì anh sẽ nói rõ :))
Trường hợp bài này thì nó giống giống với bài Find Minimum in Rotated Sorted Array rồi, chỉ modify tí thôi nên anh ngồi giải cố vẫn được.

Reply:
ok ngon. Merged :)) ĐI pv cứ giữ phong cách code này là có offer liền :))